splunk_add_on_ucc_framework/main.py

```diff
@@ -130,6 +130,13 @@ def main(argv: Optional[Sequence[str]] = None) -> int:
         help="Use old pip dependency resolver by adding flag '--use-deprecated=legacy-resolver' "
         "to pip install command.",
     )
+    build_parser.add_argument(
+        "--pip-custom-flag",
+        help="Custom flag that will be add to pip install command",
+        type=str,
+        default=False,
```

Comment:
(not in the scope of this PR but worth to think about for the future)

I am wondering if we need to specify a default value here and in `build.py::generate` function as well.

Some folks use UCC not only as a CLI but also as a library (meaning that they call `build.py::generate` from their Python scripts).